src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,53 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  max-width: #{7 * $size + 6 * $gap-size + 2 * $pading};
+  padding: #{$pading};
+  gap: #{$gap-size};
+
+  @each $day, $num in $days {
+    &--start-day--#{$day} :first-child {
+      margin-left: #{$num * $size + $gap-size * $num};
+    }
+  }
+
+  @for $monthDay from 28 through 31 {
+    &--month-length--#{$monthDay} :nth-child(n + #{$monthDay + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    position: relative;
+    box-sizing: border-box;
+    width: 100px;
+    height: 100px;
+    border: 1px solid black;
+    background-color: #eee;
+
+    &::before {
+      display: flex;
+      align-items: center;
+      text-align: center;
+      justify-content: center;
+
+      @include daySize($size);
+
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index})::before {
+        content: "#{$index}";
+      }
+    }
+    
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transition: transform 0.5s ease, background-color 0.5s ease;

```

Comment:
do not add `transition` property only to `:hover`. Your smoothness doesn't work. Move to 27 line